poc/vdaf.py

```diff
@@ -217,98 +250,98 @@ def run_vdaf(vdaf,
     """
 
     # REMOVE ME
-    test_vec = {
+    test_vec: TestVectorDict[Measurement, AggParam, AggResult] = {
         'shares': vdaf.SHARES,
         'verify_key': verify_key.hex(),
         'agg_param': agg_param,
         'prep': [],
         'agg_shares': [],
         'agg_result': None,  # set below
     }
-    type_params = vdaf.test_vec_set_type_param(test_vec)
+    type_params = vdaf.test_vec_set_type_param(cast(dict[str, Any], test_vec))
 
     out_shares = []
     for (nonce, measurement) in zip(nonces, measurements):
         assert len(nonce) == vdaf.NONCE_SIZE
 
+        # Each Client shards its measurement into input shares.
+        rand = gen_rand(vdaf.RAND_SIZE)
+        (public_share, input_shares) = \
+            vdaf.shard(measurement, nonce, rand)
+
         # REMOVE ME
-        prep_test_vec = {
+        prep_test_vec: PrepTestVectorDict[Measurement] = {
             'measurement': measurement,
             'nonce': nonce.hex(),
-            'public_share': None,  # set below
             'input_shares': [],
             'prep_shares': [[] for _ in range(vdaf.ROUNDS)],
             'prep_messages': [],
             'out_shares': [],
+            'rand': rand.hex(),
+            'public_share': vdaf.test_vec_encode_public_share(public_share).hex()
         }
-
-        # Each Client shards its measurement into input shares.
-        rand = gen_rand(vdaf.RAND_SIZE)
-        (public_share, input_shares) = \
-            vdaf.shard(measurement, nonce, rand)
-
-        # REMOVE ME
-        prep_test_vec['rand'] = rand.hex()
-        prep_test_vec['public_share'] = \
-            vdaf.test_vec_encode_public_share(public_share).hex()
         for input_share in input_shares:
             prep_test_vec['input_shares'].append(
                 vdaf.test_vec_encode_input_share(input_share).hex())
 
         # Each Aggregator initializes its preparation state.
         prep_states = []
-        outbound = []
+        outbound_prep_shares = []
```

Comment:
`mypy` does not like it when we reuse one variable name in a method in two disjoint regions with different types, so I've split uses of `outbound` as appropriate here, and done similar elsewhere.